Sources/GRPCCore/Transport/ServerTransport.swift

```diff
@@ -19,17 +19,31 @@ public protocol ServerTransport: Sendable {
   typealias Inbound = RPCAsyncSequence<RPCRequestPart>
   typealias Outbound = RPCWriter<RPCResponsePart>.Closable
 
+  /// A sequence of accepted streams to handle.
+  ///
+  /// This property is `async` because implementations will typically require ``listen()`` to be called
+  /// (and for the startup to be succesful) before a stream sequence can be returned.
+  ///
+  /// If the call to ``listen()`` throws, meaning the transport failed to start, implementations of this transport
+  /// will typically throw upon getting this property.
+  ///
+  /// Once ``listen()`` stops running, the sequence will be finished.
+  var acceptedStreams: RPCAsyncSequence<RPCStream<Inbound, Outbound>> { get async throws }
```

Comment:
I like the semantics here (i.e. throw if listen failed) but I'm not sure it's that easy to program against if you're implementing a transport. (I'm less concerned about using it because we only do that once.)

I wonder if we should change this to be a non-throwing `AsyncSequence` of events and have two events:
- `startedListening(RPCAsyncSequence<...>)`
- `failedToStartListening(any Error)` (or a better name...)

That way we can remove the `async` and `throws` from here. It also means that implementations of the transport protocol don't have to worry about e.g. multiple consumers of this property (the async sequence implementation would take care of this) or cancellation when getting the sequence (because it wouldn't be `async`). WDYT?